src/drt/src/dr/FlexDR.cpp

```diff
@@ -40,14 +44,60 @@
 #include "db/infra/frTime.h"
 #include "dr/FlexDR_conn.h"
 #include "dr/FlexDR_graphics.h"
+#include "dst/Distributed.h"
 #include "frProfileTask.h"
 #include "gc/FlexGC.h"
+#include "serialization.h"
 
 using namespace std;
 using namespace fr;
 
+BOOST_CLASS_EXPORT(RoutingJobDescription)
+
+enum class SerializationType
+{
+  READ,
+  WRITE
+};
+
+static bool serialize_worker(SerializationType type,
+                             FlexDRWorker* worker,
+                             const std::string& name)
+{
+  if (type == SerializationType::READ) {
+    std::ifstream file(name);
+    if (!file.good())
+      return false;
+    InputArchive ar(file);
+    register_types(ar);
+    ar >> *worker;
+    file.close();
+  } else {
+    std::ofstream file(name);
+    if (!file.good())
+      return false;
+    OutputArchive ar(file);
+    register_types(ar);
+    ar << *worker;
+    file.close();
+  }
+  return true;
```

Comment:
We should transition this to a switch statement instead of an if so that the compiler can warn us if we're missing a branch.

Reply:
I agree with you and that's normally what I do, but the enum has only 2 values;READ & WRITE. And I hardly doubt there are gonna be other values in the future.

Reply:
If the c++ committee would ever finish the reflection API you could just statically assert that...